test/Azure.Functions.Cli.Tests/E2E/VersionTests.cs

```diff
@@ -24,5 +27,39 @@ await CliTester.Run(new RunConfiguration
                 CommandTimeout = TimeSpan.FromSeconds(30)
             }, _output);
         }
+
+        [Fact]
+        public void CoreToolsAssemblyZipFile_ShouldParseCorrectSegment_WhenValidDownloadLinkIsProvided()
+        {
+            var fakeDownloadLink = "https://example.com/public/coretoolnumber/V4/assemblyfile.zip";
+            var releaseDetail = new CoreToolsRelease { DownloadLink = fakeDownloadLink };
+            var releaseSummary = new ReleaseSummary ("V4",releaseDetail);
+
+            var result = releaseSummary.CoreToolsAssemblyZipFile;
+
+            result.Should().Be("assemblyfile.zip"); // We expect the segment "assemblyfile.zip" based on the provided URL
+        }
+
+        [Fact]
+        public void CoreToolsAssemblyZipFile_ShouldReturnEmpty_WhenDownloadLinkIsNull()
+        {
+            var releaseDetail = new CoreToolsRelease { DownloadLink = null };
+            var releaseSummary = new ReleaseSummary("V4", releaseDetail);
+
+            var result = releaseSummary.CoreToolsAssemblyZipFile; 
+
+            result.Should().Be(string.Empty); // The result should be empty when there is no link
+        }
+
+        [Theory]
+        [InlineData("4.0.1", true)]
```

Comment:
Please add a test input which produces false as the output for the method being tested.

Reply:
@kshyju Thanks for the review.  added test input (InlineData) which produces false as the output. (By setting the value of assemblyzipfile manually because of dynamic nature of its value with regards the release).